### PR TITLE
Remove usage of jsonpointer

### PR DESCRIPTION
**Pull request by miparnisari**
*Issue #, if available:* N/A

*Description of changes:* Remove usage of [jsonpointer](https://pypi.org/project/jsonpointer/)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/maintenance.py [hunk 1]
```diff
@@ -155,7 +154,7 @@ def patch_spec(content, region):
                     jsonpatch.JsonPatch([all_patch]).apply(content, in_place=True)
                 except jsonpatch.JsonPatchConflict:
                     LOGGER.debug('Patch (%s) not applied in region %s', all_patch, region)
-                except jsonpointer.JsonPointerException:
+                except jsonpatch.JsonPointerException:
```
Comment (kddejong) [inline src/cfnlint/maintenance.py:157]: huh, weird that the exception lists switched to `jsonpointer`.  Am I missing something here?  How did you catch this issue?

Comment (miparnisari) [reply src/cfnlint/maintenance.py:157]: I just went through the list of dependencies in Scallion and noticed this package was there but in code it only appeared here, and it's not listed in setup.py

Comment (kddejong) [reply src/cfnlint/maintenance.py:157]: Wish I would have written a test for this exception

Comment (miparnisari) [reply src/cfnlint/maintenance.py:157]: It is covered by tests though. I am not sure why they passed 🤔

Comment (kddejong) [reply src/cfnlint/maintenance.py:157]: I guess that is a good sign.  I was having troubles finding this definition in jsonpatch but it must be as it works.

**Pull request merged**